quic/s2n-quic/src/tests/skip_packets.rs

```diff
@@ -0,0 +1,85 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+use super::*;
+
+#[test]
+fn optimistic_ack_mitigation() {
+    let model = Model::default();
+    model.set_delay(Duration::from_millis(50));
+    const LEN: usize = 1_000_000;
+
+    let server_subscriber = recorder::PacketSkipped::new();
+    let server_events = server_subscriber.events();
+    let client_subscriber = recorder::PacketSkipped::new();
+    let client_events = server_subscriber.events();
+    test(model, |handle| {
+        let mut server = Server::builder()
```

Comment:
Should we seed the random provider and assert an exact skip count? You could then run the test a few times with different seeds.

Reply:
Speaking of: we need to set the seeds for all of the other tests since this PR introduces non-determinism

Reply:
Addressed here https://github.com/aws/s2n-quic/pull/1986/commits/31d8f0d199a38ab4615b665d773786ff9e0d6921

>  You could then run the test a few times with different seeds.
I tested with different seeds locally. Please let me know if I interpreted your suggestion incorrectly.